hack/generated/pkg/armclient/template_client.go

```diff
@@ -0,0 +1,322 @@
+	//}
+
+	resourceName := resourceSpec.GetName()
+	names := strings.Split(resourceName, "/")
+	formattedNames := make([]string, len(names))
+	for i, name := range names {
+		formattedNames[i] = fmt.Sprintf("'%s'", name)
+	}
+
+	deployment := NewResourceGroupDeployment(atc.SubscriptionID, resourceGroup, deploymentName, resourceSpec)
+	resourceIdTemplateFunction := fmt.Sprintf("resourceId('%s', %s)", resourceSpec.GetType(), strings.Join(formattedNames, ", "))
+	deployment.Properties.Template.Outputs = map[string]Output{
+		"resourceId": {
+			Type:  "string",
+			Value: fmt.Sprintf("[%s]", resourceIdTemplateFunction),
+		},
+	}
+
+	return deployment
+}
+
+func (atc *AzureTemplateClient) BeginDeleteResource(
+	ctx context.Context,
+	id string,
+	apiVersion string,
+	status genruntime.ArmResourceStatus) error {
+
+	if id == "" {
+		return errors.Errorf("resource ID cannot be empty")
+	}
+
+	path := fmt.Sprintf("%s?api-version=%s", id, apiVersion)
+	if err := atc.RawClient.DeleteResource(ctx, path, &status); err != nil {
+		return errors.Wrapf(err, "failed deleting %s", id)
+	}
+
+	return nil
+}
+
+// HeadResource checks to see if the resource exists
+//
+// Note: this doesn't actually use HTTP HEAD as Azure Resource Manager does not uniformly implement HEAD for all
+// all resources. Also, ARM returns a 400 rather than 405 when requesting HEAD for a resource which the Resource
+// Provider does not implement HEAD. For these reasons, we use an HTTP GET
+func (atc *AzureTemplateClient) HeadResource(ctx context.Context, id string, apiVersion string) (bool, error) {
+	if id == "" {
+		return false, fmt.Errorf("resource ID cannot be empty")
+	}
+
+	idAndAPIVersion := id + fmt.Sprintf("?api-version=%s", apiVersion)
+	err := atc.RawClient.GetResource(ctx, idAndAPIVersion, nil, nil)
+	switch {
+	case IsNotFound(err):
+		return false, nil
+	case err != nil:
+		return false, err
+	default:
+		return true, nil
+	}
+}
+
+//func fillResource(de *Deployment, res *Resource) error {
+//	res.DeploymentID = de.ID
+//	if de.Properties != nil {
+//		res.ProvisioningState = de.Properties.ProvisioningState
+//	}
+//
+//	if de.Properties != nil && de.Properties.Outputs != nil {
+//		var templateOutputs map[string]TemplateOutput
+//		if err := json.Unmarshal(de.Properties.Outputs, &templateOutputs); err != nil {
+//			return err
+//		}
+//
+//		templateOutput, ok := templateOutputs["resource"]
+//		if !ok {
+//			return errors.New("could not find the resource output in the outputs map")
+//		}
+//
+//		tOutValue := templateOutput.Value
+//		res.SubscriptionID = tOutValue.SubscriptionID
+//		res.Properties = tOutValue.Properties
+//
+//		if de.Properties.OutputResources != nil && len(de.Properties.OutputResources) == 1 && de.Properties.OutputResources[0].ID != "" {
+//			// seems like this returns a more accurate ID than the resource ID function
+//			res.ID = de.Properties.OutputResources[0].ID
+//		} else {
+//			res.ID = tOutValue.ID
+//		}
+//	}
+//	return nil
+//}
```

Comment:
Should this method be restored or deleted?

Reply:
Good catch, thanks - deleted this